public/js/invertedIndex.js

```diff
@@ -0,0 +1,189 @@
+   * @memberOf InvertedIndex
+   */
+  checkIfJson(uploadedFileName) {
+    const fileExtension = this.checkFileExtension(uploadedFileName);
+    return fileExtension.toLowerCase() === 'json';
+  }
+  /**
+   * containsTitleText (it checks if the content of the file contains title and text)
+   * @returns {Boolean} - returns true if the file contains title and text or false if it does not
+   * @memberOf InvertedIndex
+   */
+  containsTitleText() {
+    for (let arrayIndex = 0; arrayIndex < this.uploadedFile.length; arrayIndex += 1) {
+      if (('title' in this.uploadedFile[arrayIndex]) && ('text' in this.uploadedFile[arrayIndex])) {
+        return true;
+      }
+    }
+    return false;
+  }
+  /**
+   * getTitlesAndTexts (it gets all the titles and texts in the uploaded file)
+   * @returns {Object} - returns an object with keys titles and texts
+   * @memberOf InvertedIndex
+   */
+  getTitlesAndTexts() {
+    const objectToReturn = [{ titles: [] }, { texts: [] }];
+
+    this.uploadedFile.forEach((e) => {
+      objectToReturn[0].titles.push(e.title.split(':')[0]);
+      objectToReturn[1].texts.push(e.text);
+    });
+
+    return objectToReturn;
+  }
+  /**
+   * getIndexedWords (it returns an array of indexed words)
+   * @returns {Array} - returns an array of indexed words
+   * @memberOf InvertedIndex
+   */
+  getIndexedWords() {
+    let uncleanIndexedWords = '';
+    const fileToIndex = this.getTitlesAndTexts();
+
+    fileToIndex[1].texts.forEach((e) => {
+      uncleanIndexedWords += `${e} `;
+    });
+
+    const cleanIndexedWords = this.cleanIndexedWords(uncleanIndexedWords);
+    // To remove the empty space at the end of the cleanIndexedWords array
+    cleanIndexedWords.pop();
+    return cleanIndexedWords;
+  }
+  /**
+   * cleanIndexedWords (it returns an array of words without commas,
+   * fullstop, spaces and also unique words)
+   * @param {string} uncleanWords
+   * @returns {Array} - an array of unique words
+   * @memberOf InvertedIndex
+   */
+  cleanIndexedWords(uncleanWords) {
+    const cleanWords = uncleanWords.replace(/\.|,/g, '').split(' ');
+    return cleanWords.filter((element, index) => {
+      return cleanWords.indexOf(element) === index;
+    });
+  }
+  /**
+   * contentToDisplay (it returns an Object of data to display to the user)
+   * @returns {Array} - an array of indexed word in a format that can be displayed to the users
+   * @memberOf InvertedIndex
+   */
+  contentToDisplay() {
+    const displayIndexedWords = [];
+    const fileToDisplay = this.getTitlesAndTexts();
+    const indexedWords = this.getIndexedWords();
+    displayIndexedWords.push(indexedWords);
+
+    for (let arrayIndex = 0; arrayIndex < fileToDisplay[1].texts.length; arrayIndex += 1) {
+      const temporaryData = [];
+      const newIndexedWords = this.cleanIndexedWords(fileToDisplay[1].texts[arrayIndex]);
+
+      indexedWords.forEach((e) => {
+        if (newIndexedWords.includes(e)) {
```

Comment:
This `if-else` block can be replaced by one line of code. `temporaryData.push(newIndexedWords.includes(e))`. Or you can also save the value of `newIndexedWords.includes(e)` in a variable `isIncluded` for example and call `temporaryData.push(isIncluded)`.